en/release-notes/upgrade-guide-v3.0.md

```diff
@@ -314,14 +431,138 @@ Notice from the above two figures, there is a big difference between v2.4.3 and
 
-## TypeScript Reference Tutorial
+## Upgrade FAQ
+
+### After the upgrade, the project script shows errors when opening in VS Code with operations such as binding component definitions
+
+Cocos Creator 3.x enables Strict Mode for TypeScript, which will review the code more strictly and eliminate any problems that may occur due to negligence during the development process.
+
+If you don't want to use strict mode, you can check **Enable loose mode** in **Project -> Project Settings -> Scripting** in the top menu bar of Creator. As a reminder, we do not encourage turning off strict mode, as strict null checking can reduce some low-level errors in the code runtime.
+
+For writing specifications in strict mode, you can refer to the official case [Taxi Game 3D](https://github.com/cocos-creator/tutorial-taxi-game).
+
+### `Action` actions are all disabled
+
+Because Cocos Creator 3.x removes the `Action` action system and uses the `Tween` tweening system instead.
+
+### Modifying `size` and `anchor` of 2D nodes does not work
+
+You need to get the UITransform component on the node first, and then use the corresponding interface, e.g.
+
+```typescript
+const uiTrans = node.getComponent(UITransform)! ;
+uiTrans.anchorX = 0.5;
+uiTrans.setContentSize(size);
+```
+
+### Modifying the `color` of a 2D node does not work
+
+You need to get the rendering component on the node (e.g. Sprite component) first, and then use the corresponding interface, e.g.
+
+```typescript
+const uiColor = node.getComponent(Sprite)! ;
+uiColor.color = color(255,255,255);
+```
+
+### Modifying `skew` for 2D nodes does not work
+
+As of v3.0, the `skew` interface has been removed.
+
+### Grouping is not available, but there are still grouping settings (Layers) in the Creator's project settings panel
+
+The `group` group management in v2.x has been changed to `Layer` since v3.0, as shown below. In v2.x the group name is obtained from `node.group`, while in v3.x the group value is obtained from `node.layer` as **group value**, and the group value is set in exponential power of 2.
+
+![update-setting](update-setting.png)
+
+The layer value of User Layer 0 is: 2<sup>0</sup> = 1.<br>
+The layer value of User Layer 1 is: 2<sup>1</sup> = 2.<br> 
+The layer value of User Layer 6 is: 2<sup>6</sup> = 64.
+
+### The sibling node set by `zIndex` is invalid
+
+As of v3.0 the `zIndex` interface has been removed, if you need to reorder the node tree please use the `priority` method instead.
+
+### The script mounted on the node is not available via `getComponent()`.
+
+Please look up the class name of the corresponding script, not the script name, because in v3.x script components are based on the class name defined in the script, not the script name. There are often problems with scripts not being found due to case. Please refer to [Create Script](...) for more details. /scripting/setup.md).
```

Comment:
```suggestion
Please look up the class name of the corresponding script, not the script name. v3.x script components are based on the class name defined in the script, not the script name. There are often problems with scripts not being found due to letter cases. Please refer to the [Create Script](.../scripting/setup.md) documentation for more details.
```

Reply:
There was a URL formatting error. I fixed it, but please double-check.

Reply:
Fixed.